src/pages/signup/signup.test.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import {
+  fireEvent,
+  render,
+  screen,
+  waitFor,
+} from '@testing-library/react';
+import SignUp from 'pages/signup';
+import userData from 'users.json';
+import bcrypt from 'bcryptjs';
+
+const handleInput = (selector, value) => {
+  fireEvent.change(selector, {
+    target: {
+      value,
+    },
+  });
+};
+
+const fireSubmit = () => {
+  fireEvent.click(screen.getByRole('button', { name: 'Sign up' }));
+};
+
+const testCases = (type, dataCases) => {
+  const title = `Render ${type} $field error message`;
+  test.each(dataCases)(title, async ({ field, data, error }) => {
+    render(<SignUp />);
+
+    const inpField = screen.getByLabelText(field);
+    const inpData = data;
+
+    handleInput(inpField, inpData);
+    fireSubmit();
+
+    await waitFor(() => {
+      expect(screen.getByText(error)).toBeInTheDocument();
+    });
+  });
+};
+
+describe('Sign up functionality', () => {
+  beforeAll(() => {
+    localStorage.setItem('users', JSON.stringify(userData));
+  });
+
+  const invalidCases = [
```

Comment:
The cases should be in the test case, splitting it into object else where will make your partner have to find the object whenever he/ she  read the test code

Reply:
Noted, I will update them accordingly